### PR TITLE
BVTCK-158 Add a test for cascading validation with 2 inner extractions

### PR DESCRIPTION
**Pull request by gsmet**
* https://hibernate.atlassian.net/browse/BVTCK-158

### CODE REVIEW

File: tests/src/main/java/org/hibernate/beanvalidation/tck/tests/validation/graphnavigation/containerelement/NestedCascadingOnContainerElementsTest.java [hunk 1]
```diff
@@ -82,6 +86,41 @@ public void testNestedValid() {
...
+
+		assertNumberOfViolations( constraintViolations, 0 );
+
+		constraintViolations = validator.validate( AddressBook.invalid() );
+
+		assertThat( constraintViolations ).containsOnlyViolations(
+				violationOf( NotBlank.class )
+						.withPropertyPath( pathWith()
+								.property( "addressesPerTypePerPerson" )
+								.containerElement( "<map value>", true, "Firstname Lastname", null, Map.class, 1 )
```
Comment (gunnarmorling) [inline tests/src/main/java/org/hibernate/beanvalidation/tck/tests/validation/graphnavigation/containerelement/NestedCascadingOnContainerElementsTest.java]: Shouldn't this one be on the map key actually?

**Comment by gunnarmorling** [comment]
@gsmet I'm puzzled by that one test. Could you also rebase the PR to the latest master?

File: tests/src/main/java/org/hibernate/beanvalidation/tck/tests/validation/graphnavigation/containerelement/NestedCascadingOnContainerElementsTest.java (hunk 1 above)
Comment (gsmet) [inline tests/src/main/java/org/hibernate/beanvalidation/tck/tests/validation/graphnavigation/containerelement/NestedCascadingOnContainerElementsTest.java]: It is :).

It's a nested Map with 2 levels: it's the value on the first level and the key for the second. See the line below, the `typeArgumentIndex` is 0.

(btw, unrelated, but I force pushed a minor cleanup as there was `@Valid` annotations wandering around in the Map initialization code)

Comment (gunnarmorling) [inline tests/src/main/java/org/hibernate/beanvalidation/tck/tests/validation/graphnavigation/containerelement/NestedCascadingOnContainerElementsTest.java]: Ah yeah, true that!

**Comment by gsmet** [comment]
@gunnarmorling rebased!

**Comment by gunnarmorling** [comment]
Sweet, down to zero pending PRs, jippie :) Thanks, @gsmet!

**Pull request merged**